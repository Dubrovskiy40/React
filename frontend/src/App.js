import './App.css'
import Footer from './components/Footer/Footer';
import Header from './components/Header/Header';
import Profile from './components/Profile/Profile';
import { BrowserRouter as Router, Route, useHistory } from "react-router-dom";
import Navbar from './components/Navbar/Navbar';
import Dialogs from './components/Dialogs/Dialogs';
import Main from "./components/Main/Main";
import Comments from "./components/ShowAPI/Comments";
import BackendDate from './components/BackendDate/BackendDate';
import PrivateRoute from './components/SecureRoute/SecureRoute';

function App(props) {
  const history = useHistory();

  return (
      <Router history={history}>
        <div className="app__wrap">
          <Header />
          <Navbar />
          <div className='app__content'>
            <Route exact path='/' render={ () => <Main />} />
            <PrivateRoute path='/profile' render={ () => <Profile />} />
            <PrivateRoute path='/dialogs' render={ () => <Dialogs />} />
            <PrivateRoute path='/api' render={ () => <Comments />} />
            <Route path='/backendDate' render={ () => <BackendDate />} />
          </div>
        </div>
        <Footer />
      </Router>
  );
};

export default (App);