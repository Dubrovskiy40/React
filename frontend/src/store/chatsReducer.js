import { 
    ADD_CHAT, 
    DELETE_CHAT,
    CREATE_CHAT
} from "./types/chatsTypes"

const initialState = { activeId: 0,
    chats: [

    ]
};
const chatsReducer = (state = initialState, {type, payload}) => {
    switch (type) {
        case ADD_CHAT:
            return { ...state, activeId: payload };
        case DELETE_CHAT:
            return {...state, chats: state.chats.filter((chat) => chat.id !== payload)};
        case CREATE_CHAT:
            return {...state, chats: [...state.chats, {id: state.chats.length, name: payload}]};  
        default:
            return state;
    }
};

export default chatsReducer;