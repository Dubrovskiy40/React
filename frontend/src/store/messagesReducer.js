import {
  ADD_MESSAGE,
  DELETE_MESSAGE,
  DELETE_MESSAGELIST,
} from "./types/messageTypes";

const initialState = {
  0: [
    {
      id: 0,
      text: "Темная сторона захватила тебя",
      author: "Чубака",
    },
    {
      id: 1,
      text: "летим туда, там столько вкусного",
      author: "Штурмовик",
    },
  ],
  1: [
    {
      id: 0,
      text: "Принцесса в опасности!",
      author: "Люк",
    },
    {
      id: 1,
      text: "Крылья, лапы... главное хвост!",
      author: "Ящерица",
    },
  ],
  2: [
    {
      id: 0,
      text: "Необходимо лететь на другую планету",
      author: "R2D2",
    },
    {
      id: 1,
      text: "Необходимо захватить эту планету",
      author: "Ваш Дарт",
    },
  ],
  3: [
    {
      id: 0,
      text: "Предлагаю объединить наши силы",
      author: "Беженцы",
    },
    {
      id: 1,
      text: "Ситх предлагает сдаться",
      author: "Ситх",
    },
  ],
  4: [
    {
      id: 0,
      text: "У меня есть кое что, что поднимет тебе настроение и залечит раны",
      author: "Торговец сладостей",
    },
    {
      id: 1,
      text: "Жду, номер 42",
      author: "Незнакомка",
    },
  ],
};

const messagesReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case ADD_MESSAGE:
      return {
        ...state,
        [payload.chatId]: [
          ...state[payload.chatId],
          {
            id: state[payload.chatId].length,
            ...payload.message,
          },
        ],
      };

    case DELETE_MESSAGE:
      return {
        ...state,
        [payload.chatId]: state[payload.chatId].filter(
          (message) => message.id !== payload.id
        ),
      };

    case DELETE_MESSAGELIST:
      return {
        ...state,
        [payload]: [],
      };
    default:
      return state;
  }
};

export default messagesReducer;
