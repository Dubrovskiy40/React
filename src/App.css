.app__wrap {
  margin: 0 auto;
  display: grid;
  width: auto;
  grid-template-rows: 60px 1fr;
  grid-template-columns: 3fr 9fr;
  grid-template-areas:
      "h h"
      "n m"
      "f f";
}
.app__content {
  grid-area: m;
    background-color: rgb(161, 156, 156);
    min-height: calc(100vh - 60px - 60px);
}

.btn {
  margin: 10px 20px;
  
}
