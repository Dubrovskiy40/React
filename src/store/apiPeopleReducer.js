import { 
  GET_HEROES,
  DELETE_HEROES,
  ADD_HEROES
} from './types/apiPeopleTypes';

const initialState = [];

const apiPeopleReducer = (state = initialState, { type, payload }) => {
  switch (type) {

    case GET_HEROES:
      return [...payload];

    case DELETE_HEROES:
      return state.filter((hero, index) => index !== payload);

    case ADD_HEROES:
      return [payload, ...state];

    default:
      return state;
  }
};

export default apiPeopleReducer;
