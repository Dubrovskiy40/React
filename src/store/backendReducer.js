import { SET_TEST } from "./types/backendTypes";

const initialState = null;

const backendReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case SET_TEST:
      return payload;
    default:
      return state;
  }
};

export default backendReducer;