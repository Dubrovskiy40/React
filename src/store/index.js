import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import chats from './chatsReducer';
import profile from './profileReducer';
import modalMessage from './modalMessageReducer';
import messages from './messagesReducer';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import apiCommentsReducer from './apiCommentsReducer';
import { loadReducer } from './loadReducer';

const persistConfig = {
    key: 'socialNetwork',
    storage: storage,
    blacklist: ['modalMessage'],
}

const rootReducer = combineReducers({
    profile,
    chats,
    modalMessage,
    messages,
    apiCommentsReducer,
    loadReducer
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(persistedReducer, composeWithDevTools(applyMiddleware(thunk)));

export const persistor = persistStore(store);