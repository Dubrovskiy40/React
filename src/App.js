import { useState, useEffect } from 'react';
import './App.css'
import Form from './components/Form/Form';
import Footer from './components/Footer/Footer';
import HeaderNav from './components/HeaderNav/HeaderNav';
import MainContent from './components/MainContent/MainContent';
import { withTheme } from '@material-ui/core/styles';

import Chats from './components/Chats/Chats';
import Profile from './components/Profile/Profile';

import {
  BrowserRouter as Router,
  Route,
  Link
} from "react-router-dom";


let answerBot = '';

function App() {
  const [messageList, setMessageList] = useState([]);
  const [messageText, setMessageText] = useState('');
  const [messageAuthor, setMessageAuthor] = useState('');

  useEffect(() => {
    const last = messageList.slice(-1);
    setTimeout(() => {
      if (last.length) {
        let answerAuthor = last[0].author;
        let answerText = last[0].text;
        answerBot = `Принял сообщение от ${answerAuthor} с текстом: ${answerText}`;
        console.log('Ответ робота:', answerBot);
      }
    }, 1500);
    setMessageText('');
    setMessageAuthor('');
  }, [messageList]);

  return (
    <>
      <Router>
        <HeaderNav />
        <MainContent />
        <div className='wrap'>
            <Route exact path="/profile" component={Profile} />
            <Route exact path="/chats" component={Chats} />
            <Route exact path="/"><Link to="/chats">Список чатов</Link></Route>
            <Route exact path="/"><Link to="/profile">Мой Профиль</Link></Route>
            <Route exact path="/chats/*" component={Form}>
              <Form messageList={messageList}
                messageText={messageText}
                fooText={setMessageText} 
                messageAuthor={messageAuthor} 
                fooAuthor={setMessageAuthor}
                setML={setMessageList}
                answer={answerBot}
              />
            </Route>
          </div>
        <Footer />
      </Router>
    </>
  );
};

export default withTheme(App);