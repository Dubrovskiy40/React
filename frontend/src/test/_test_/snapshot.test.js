import "@testing-library/react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import App from "../../App";
import { store, persistor } from "../../store";
import { PersistGate } from 'redux-persist/integration/react'

describe('snapshot test', () => {
    it ('test reducer correct state after USER_LOGIN', () => {
        const component = render(
            <Provider store = { store }>
                <PersistGate loading={null} persistor={persistor}>
                    <App />
                </PersistGate>
            </Provider>, 
        );
        expect(component).toMatchSnapshot();
    });
});
