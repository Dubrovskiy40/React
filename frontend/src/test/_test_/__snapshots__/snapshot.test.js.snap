// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test test reducer correct state after USER_LOGIN 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app__wrap"
      >
        <header
          class="header"
        >
          <img
            alt="logo"
            class="header__img"
            src="logo.png"
          />
          <h2>
            –°–æ—Ü–∏–∞–ª—å–Ω–∞—è —Å–µ—Ç—å –ø—Ä–∏–Ω–∞–¥–ª–µ–∂–∏—Ç: 
          </h2>
        </header>
        <nav
          class="nav"
        >
          <div
            class="item"
          >
            <a
              aria-current="page"
              class="activeLink"
              href="/"
            >
              –ì–ª–∞–≤–Ω–∞—è
            </a>
          </div>
          <div
            class="item"
          >
            <a
              href="/profile"
            >
              –ü—Ä–æ—Ñ–∏–ª—å
            </a>
          </div>
          <div
            class="item active"
          >
            <a
              href="/dialogs"
            >
              –°–æ–æ–±—â–µ–Ω–∏—è
            </a>
          </div>
          <div
            class="item active"
          >
            <a
              href="/api"
            >
              API
            </a>
          </div>
        </nav>
        <div
          class="app__content"
        >
          <div
            class="main__wrap"
          >
            <h2>
              –°–∞–º–∞—è –≥–ª–∞–≤–Ω–∞—è —Å—Ç—Ä–∞–Ω–∏—Ü–∞ –≤–æ –≤—Å–µ–ª–µ–Ω–Ω–æ–π
            </h2>
            <img
              alt="title__img"
              class="main__img"
              src="stormtrooper.png"
            />
            <div>
              <form
                class="wrap"
              >
                <input
                  class="inp"
                  placeholder="email"
                  type="email"
                  value=""
                />
                <input
                  class="inp"
                  placeholder="password"
                  type="password"
                  value=""
                />
                <button
                  class="btn"
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer"
      >
        <div>
          –í—Å–µ –ø—Ä–∞–≤–∞ –∑–∞—â–∏—Ç—ãüòÜ
        </div>
        <h4>
          Dubrovskiy
        </h4>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="app__wrap"
    >
      <header
        class="header"
      >
        <img
          alt="logo"
          class="header__img"
          src="logo.png"
        />
        <h2>
          –°–æ—Ü–∏–∞–ª—å–Ω–∞—è —Å–µ—Ç—å –ø—Ä–∏–Ω–∞–¥–ª–µ–∂–∏—Ç: 
        </h2>
      </header>
      <nav
        class="nav"
      >
        <div
          class="item"
        >
          <a
            aria-current="page"
            class="activeLink"
            href="/"
          >
            –ì–ª–∞–≤–Ω–∞—è
          </a>
        </div>
        <div
          class="item"
        >
          <a
            href="/profile"
          >
            –ü—Ä–æ—Ñ–∏–ª—å
          </a>
        </div>
        <div
          class="item active"
        >
          <a
            href="/dialogs"
          >
            –°–æ–æ–±—â–µ–Ω–∏—è
          </a>
        </div>
        <div
          class="item active"
        >
          <a
            href="/api"
          >
            API
          </a>
        </div>
      </nav>
      <div
        class="app__content"
      >
        <div
          class="main__wrap"
        >
          <h2>
            –°–∞–º–∞—è –≥–ª–∞–≤–Ω–∞—è —Å—Ç—Ä–∞–Ω–∏—Ü–∞ –≤–æ –≤—Å–µ–ª–µ–Ω–Ω–æ–π
          </h2>
          <img
            alt="title__img"
            class="main__img"
            src="stormtrooper.png"
          />
          <div>
            <form
              class="wrap"
            >
              <input
                class="inp"
                placeholder="email"
                type="email"
                value=""
              />
              <input
                class="inp"
                placeholder="password"
                type="password"
                value=""
              />
              <button
                class="btn"
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      <div>
        –í—Å–µ –ø—Ä–∞–≤–∞ –∑–∞—â–∏—Ç—ãüòÜ
      </div>
      <h4>
        Dubrovskiy
      </h4>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
